pkg/integrations/v2/app_o11y_receiver/config/config.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import "time"
+
+const (
+	// DefaultRateLimitingRPS is the default value of Requests Per Second
+	// for ratelimiting
+	DefaultRateLimitingRPS = 100
+	// DefaultRateLimitingBurstiness is the default burstiness factor of the
+	// token bucket algorigthm
+	DefaultRateLimitingBurstiness = 50
+	// DefaultMaxPayloadSize is the max paylad size in bytes
+	DefaultMaxPayloadSize = 5e6
+)
+
+// DefaultConfig holds the default configuration of the receiver
+var DefaultConfig = AppO11yReceiverConfig{
+	// Default JS agent port
+	CORSAllowedOrigins: []string{},
+	RateLimiting: RateLimitingConfig{
+		Enabled:    false,
+		RPS:        DefaultRateLimitingRPS,
+		Burstiness: DefaultRateLimitingBurstiness,
+	},
+	MaxAllowedPayloadSize: DefaultRateLimitingRPS,
+	Server: ServerConfig{
+		Host: "0.0.0.0",
+		Port: 8080,
+	},
+	TracesInstance:  "",
+	LogsInstance:    "",
+	LogsLabels:      map[string]string{},
+	LogsSendTimeout: 2000,
+	SourceMaps: SourceMapConfig{
+		Download:            false,
+		DownloadFromOrigins: []string{"*"},
+		DownloadTimeout:     time.Duration(1000000),
+		FileSystem:          nil,
+	},
+}
+
+// ServerConfig holds the receiver http server configuration
+type ServerConfig struct {
+	Host string `yaml:"host,omitempty"`
+	Port int    `yaml:"port,omitempty"`
+}
+
+// RateLimitingConfig holds the configuration of the rate limiter
+type RateLimitingConfig struct {
+	Enabled    bool    `yaml:"enabled,omitempty"`
+	RPS        float64 `yaml:"rps,omitempty"`
+	Burstiness int     `yaml:"burstiness,omitempty"`
+}
+
+// SourceMapFileLocation holds sourcemap location on file system
+type SourceMapFileLocation struct {
+	Path               string `yaml:"path"`
+	MinifiedPathPrefix string `yaml:"minified_path_prefix,omitempty"`
+}
+
+// SourceMapConfig configure source map locations
+type SourceMapConfig struct {
+	Download            bool                    `yaml:"download"`
+	DownloadFromOrigins []string                `yaml:"download_origins,omitempty"`
+	DownloadTimeout     time.Duration           `yaml:"download_timeout,omitempty"`
+	FileSystem          []SourceMapFileLocation `yaml:"filesystem,omitempty"`
+}
+
+// AppO11yReceiverConfig is the configuration struct of the
+// integration
+type AppO11yReceiverConfig struct {
+	CORSAllowedOrigins    []string           `yaml:"cors_allowed_origins,omitempty"`
+	RateLimiting          RateLimitingConfig `yaml:"rate_limiting,omitempty"`
```

Comment:
nit: make this a pointer with its own UnmarshalYAML implementation, so:

1. You don't need the enabled field (if it's not defined, it's not enabled)
2. You can more easily default the rate limit when it is provided

Reply:
It should be enabled by default, it's a bad idea to go without, even accidentally :thinking:  Maybe we should just remove the `enabled` flag